src/factory.ts

```diff
@@ -66,14 +63,74 @@ export class CommentFactory<T = any> implements ICommentFactory<T> {
       type: 'reply'
     };
   }
-
   readonly type: string;
   readonly targetFactory: (target: T) => PartialJSONValue;
 }
 
-export namespace CommentFactory {
+export class CellCommentFactory extends ACommentFactory {
+  constructor() {
+    super({
+      type: 'cell',
+      targetFactory: (cell: Cell) => {
```

Comment:
instead of having a `targetFactory` passed in, `ACommentFactory` should have an abstract method called `targetToJSON` or something.